pkg/plugins/external/external_test.go

```diff
@@ -248,4 +263,257 @@ var _ = Describe("Run external plugin using Scaffold", func() {
 
 		})
 	})
+
+	Context("with successfully getting flags from external plugin", func() {
+		var (
+			pluginFileName string
+			args           []string
+			flagset        *pflag.FlagSet
+
+			// Make an array of flags to represent the ones that should be returned in these tests
+			flags = getFlags()
+
+			checkFlagset func()
+		)
+		BeforeEach(func() {
+			outputGetter = &mockValidFlagOutputGetter{}
+			currentDirGetter = &mockValidOsWdGetter{}
+
+			pluginFileName = "myexternalplugin.sh"
+			args = []string{"--captain", "black-beard", "--sail"}
+			flagset = pflag.NewFlagSet("test", pflag.ContinueOnError)
+
+			checkFlagset = func() {
+				Expect(flagset.HasFlags()).To(BeTrue())
+
+				for _, flag := range flags {
+					Expect(flagset.Lookup(flag.Name)).NotTo(BeNil())
+					// we parse floats as float64 Go type so this check will account for that
+					if flag.Type != "float" {
+						Expect(flagset.Lookup(flag.Name).Value.Type()).To(Equal(flag.Type))
+					} else {
+						Expect(flagset.Lookup(flag.Name).Value.Type()).To(Equal("float64"))
+					}
+					Expect(flagset.Lookup(flag.Name).Usage).To(Equal(flag.Usage))
+					Expect(flagset.Lookup(flag.Name).DefValue).To(Equal(flag.Default))
+				}
+			}
+		})
+
+		It("should successfully bind external plugin specified flags for `init` subcommand", func() {
+			sc := initSubcommand{
+				Path: pluginFileName,
+				Args: args,
+			}
+
+			sc.BindFlags(flagset)
+
+			checkFlagset()
+		})
+
+		It("should successfully bind external plugin specified flags for `create api` subcommand", func() {
+			sc := createAPISubcommand{
+				Path: pluginFileName,
+				Args: args,
+			}
+
+			sc.BindFlags(flagset)
+
+			checkFlagset()
+		})
+
+		It("should successfully bind external plugin specified  flags for `create webhook` subcommand", func() {
+			sc := createWebhookSubcommand{
+				Path: pluginFileName,
+				Args: args,
+			}
+
+			sc.BindFlags(flagset)
+
+			checkFlagset()
+		})
+
+		It("should successfully bind external plugin specified flags for `edit` subcommand", func() {
+			sc := editSubcommand{
+				Path: pluginFileName,
+				Args: args,
+			}
+
+			sc.BindFlags(flagset)
+
+			checkFlagset()
+		})
+	})
+
+	Context("with failure to get flags from external plugin", func() {
+		var (
+			pluginFileName string
+			args           []string
+			flagset        *pflag.FlagSet
+			usage          string
+			checkFlagset   func()
+		)
+		BeforeEach(func() {
+			outputGetter = &mockInValidOutputGetter{}
+			currentDirGetter = &mockValidOsWdGetter{}
+
+			pluginFileName = "myexternalplugin.sh"
+			args = []string{"--captain", "black-beard", "--sail"}
+			flagset = pflag.NewFlagSet("test", pflag.ContinueOnError)
+			usage = "Kubebuilder could not validate this flag with the external plugin. Consult the external plugin documentation for more information."
+
```

Comment:
You cna test the lint locally with make lint
also some silly issues can be fixed with make lint-fix 

however, in this case the error is the size of the line which is too big so it will not be fixed with the above option

Reply:
All lint issues should be fixed in https://github.com/kubernetes-sigs/kubebuilder/pull/2693/commits/9648e34decca255094c2a4b1934d92ae014caa78